data_steward/tools/snapshot_by_query.py

```diff
@@ -118,7 +130,20 @@ def copy_tables_to_new_dataset(project_id, dataset_id, snapshot_dataset_id):
     """
     copy_table_job_ids = []
     client = bq.get_client(project_id)
+    destination_tables = list_all_table_ids(snapshot_dataset_id)
     for table_id in list_all_table_ids(dataset_id):
+        LOGGER.info(
+            f" Copying {dataset_id}.{table_id} to {snapshot_dataset_id}.{table_id}"
+        )
+        if table_id not in destination_tables:
+            try:
+                fields = resources.fields_for(table_id)
+                bq.create_tables(
+                    client, project_id,
+                    [f'{project_id}.{snapshot_dataset_id}.{table_id}'], False,
+                    [fields])
+            except RuntimeError:
+                LOGGER.info(f'Unable to find schema for {table_id}')
```

Comment:
> Looks good! Let me know if you've already tested this on a dataset with fitbit tables. I added a suggestion for tables not in curation. Let me know your thoughts.

yes, I did run the test with a dataset containing Fitbit tables and they worked fine.

Reply:
> If the schema cannot be found, eg. if we are copying tables outside of curation (data_browser/cohort_builder etc.), maybe it's better to copy the table as is?

I think so doing that we have to modify other code as well. i.e for get_copy query have to ignore these tables as these were already copied. I think we can look into the optimization of this whole script to use BiQuery client instead of bq_utils in a different ticket.

Reply:
Oh right, yeah if it's already being copied then this is fine.